core/src/main/java/com/linecorp/armeria/common/util/AbstractThreadFactory.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import io.netty.util.concurrent.DefaultThreadFactory;
+
+/**
+ * A skeletal {@link ThreadFactory} implementation.
+ */
+public abstract class AbstractThreadFactory implements ThreadFactory {
```

Comment:
How about:

- making this class package-private
- making all protected members package-private
- removing all deprecated members (Note that we must keep the deprecated methods in `EventLoopThreadFactory`.)

...? No need to keep the deprecated members for a package-local class.

Reply:
Removed all deprecated constructor and used package-private constructor for deprecated constructor in `EventLoopThreadFactory`. PTAL :pray: